File: 3010-divide-an-array-into-subarrays-with-minimum-cost-i/3010-divide-an-array-into-subarrays-with-minimum-cost-i.cpp
class Solution {
public:
    int minimumCost(vector<int>& nums) {
    int n = nums.size();
    int min_sum = INT_MAX;

    for (int i = 1; i <= n - 2; i++) {
        for (int j = i + 1; j <= n - 1; j++) {
            min_sum = min(min_sum, nums[0] + nums[i] + nums[j]);
        }
    }
    return min_sum;
    }
};

File: 3010-divide-an-array-into-subarrays-with-minimum-cost-i/3010-divide-an-array-into-subarrays-with-minimum-cost-i.rs
impl Solution {
    pub fn minimum_cost(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut ans = i32::MAX;

        for i in 1..=n - 2 {
            for j in i + 1..=n - 1 {
                let cost = nums[0] + nums[i] + nums[j];
                ans = ans.min(cost);
            }
        }

        ans
    }
}
